pkg/connections/connections.go

```diff
@@ -0,0 +1,331 @@
+				if err != nil {
+					fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+				}
+			}()
+
+			cds, ssTCP, ssUDP, err := cc.createSSOutputFromNode(debugPod, node)
+			if err != nil {
+				return err
+			}
+			nLock.Lock()
+			defer nLock.Unlock()
+			ssTCPLine := fmt.Sprintf("node: %s\n%s\n", node.Name, string(ssTCP))
+			ssUDPLine := fmt.Sprintf("node: %s\n%s\n", node.Name, string(ssUDP))
+
+			nodesComDetails = append(nodesComDetails, cds...)
+			ssOutTCP = append(ssOutTCP, []byte(ssTCPLine)...)
+			ssOutUDP = append(ssOutUDP, []byte(ssUDPLine)...)
+			return nil
+		})
+	}
+
+	err = g.Wait()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	cleanedComDetails := types.CleanComDetails(nodesComDetails)
+	ssComMat := types.ComMatrix{Matrix: cleanedComDetails}
+
+	return &ssComMat, ssOutTCP, ssOutUDP, nil
+}
+
+func (cc *ConnectionCheck) WriteSSRawFiles(ssOutTCP, ssOutUDP []byte) error {
+	tcpFile, err := os.OpenFile(path.Join(cc.destDir, "raw-ss-tcp"), os.O_TRUNC|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer tcpFile.Close()
+
+	udpFile, err := os.OpenFile(path.Join(cc.destDir, "raw-ss-udp"), os.O_TRUNC|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer udpFile.Close()
+
+	_, err = tcpFile.Write(ssOutTCP)
+	if err != nil {
+		return fmt.Errorf("failed writing to file: %s", err)
+	}
+
+	_, err = udpFile.Write(ssOutUDP)
+	if err != nil {
+		return fmt.Errorf("failed writing to file: %s", err)
+	}
+
+	return nil
+}
+
+func (cc *ConnectionCheck) createSSOutputFromNode(debugPod *corev1.Pod, node *corev1.Node) ([]types.ComDetails, []byte, []byte, error) {
+	ssOutTCP, err := cc.podUtils.RunCommandOnPod(debugPod, []string{"/bin/sh", "-c", "ss -anpltH"})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	ssOutUDP, err := cc.podUtils.RunCommandOnPod(debugPod, []string{"/bin/sh", "-c", "ss -anpluH"})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	ssOutFilteredTCP := filterEntries(splitByLines(ssOutTCP))
+	ssOutFilteredUDP := filterEntries(splitByLines(ssOutUDP))
+
+	tcpComDetails := cc.toComDetails(debugPod, ssOutFilteredTCP, "TCP", cc.nodesRolesMap[node.Name])
+	udpComDetails := cc.toComDetails(debugPod, ssOutFilteredUDP, "UDP", cc.nodesRolesMap[node.Name])
+
+	res := []types.ComDetails{}
+	res = append(res, udpComDetails...)
+	res = append(res, tcpComDetails...)
+
+	return res, ssOutTCP, ssOutUDP, nil
+}
+
+func splitByLines(bytes []byte) []string {
+	str := string(bytes)
+	return strings.Split(str, "\n")
+}
+
+func (cc *ConnectionCheck) toComDetails(debugPod *corev1.Pod, ssOutput []string, protocol string, nodeRole []string) []types.ComDetails {
+	res := make([]types.ComDetails, 0)
+
+	for _, ssEntry := range ssOutput {
+		cd := parseComDetail(ssEntry)
+
+		name, err := cc.getContainerName(debugPod, ssEntry)
+		if err != nil {
+			log.Debugf("failed to identify container for ss entry: %serr: %s", ssEntry, err)
+		}
+
+		for _, role := range nodeRole {
```

Comment:
I think in SNO we should have one role only - master, to avoid duplicating the matrix

Reply:
i think we are are calling CleanComDetails after the toComDetails and with this we are avoiding the duplicates

Reply:
lets use the first role for now and we can change that in a following PR